server/game/cards/locations/04/riverrun.js

```diff
@@ -0,0 +1,35 @@
+const DrawCard = require('../../../drawcard.js');
+
+class Riverrun extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onCardPowerChanged: (event, card, power) => {
+                    if(!this.kneeled
+                       && card.getType() === 'character' && card.hasTrait('House Tully')
+                       && power > 0) {
+                        this.tullyCharacter = card;
+
+                        return true;
+                    }
+                    return false;
+                }
+            },
+            handler: () => {
+                if(!this.tullyCharacter) {
+                    return false;
+                }
+                this.controller.kneelCard(this);
```

Comment:
The new cost framework is very very very cool. Thanks for pointing me to it.

Fixed this PR so that it uses it.